Mixed cluster tests failing in release builds due to feature flag

We have an issue which I believe is fallout from #77626. The root issue is that we are passing a feature flag, but when doing BWC tests, the BWC distribution isn't built in release mode.

```
»  Caused by: java.lang.IllegalArgumentException: es.index_mode_feature_flag_registered is only supported in non-snapshot builds
»  	at org.elasticsearch.index.IndexSettings.<clinit>(IndexSettings.java:351) ~[elasticsearch-7.16.0-SNAPSHOT.jar:7.16.0-SNAPSHOT]
»  	... 13 more
```

So in this case we are testing 8.0.0 against 7.16.0-SNAPSHOT. We pass `-Dbuild.snapshot=true` when bulding `master` (8.0.0) but not when building the BWC snapshot version (`7.x` or `7.16` in this case).

What we _should_ do I suppose depends on what we're trying to test here. Our options are:

1. We carry forward the `-Dbuild.snapshot` flag to the BWC builds so we are testing mixed "released" versions.
2. We only enable the feature flag on `master` nodes and omit it when configuring the BWC nodes.

--- a/qa/mixed-cluster/build.gradle
+++ b/qa/mixed-cluster/build.gradle
@@ -39,9 +39,6 @@ for (Version bwcVersion : BuildParams.bwcVersions.wireCompatible) {
 
       setting 'path.repo', "${buildDir}/cluster/shared/repo/${baseName}"
       setting 'xpack.security.enabled', 'false'
-      if (BuildParams.isSnapshotBuild() == false) {
-        systemProperty 'es.index_mode_feature_flag_registered', 'true'
-      }
     }
   }
 
@@ -53,10 +50,16 @@ for (Version bwcVersion : BuildParams.bwcVersions.wireCompatible) {
       // Getting the endpoints causes a wait for the cluster
       println "Test cluster endpoints are: ${-> testClusters."${baseName}".allHttpSocketURI.join(",")}"
       println "Upgrading one node to create a mixed cluster"
+      if (BuildParams.isSnapshotBuild() == false) {
+        testClusters."${baseName}".nodes."${baseName}-0".systemProperty 'es.index_mode_feature_flag_registered', 'true'
+      }
       testClusters."${baseName}".nextNodeToNextVersion()
       // Getting the endpoints causes a wait for the cluster
       println "Upgrade complete, endpoints are: ${-> testClusters."${baseName}".allHttpSocketURI.join(",")}"
       println "Upgrading another node to create a mixed cluster"
+      if (BuildParams.isSnapshotBuild() == false) {
+        testClusters."${baseName}".nodes."${baseName}-1".systemProperty 'es.index_mode_feature_flag_registered', 'true'
+      }
       testClusters."${baseName}".nextNodeToNextVersion()
 
       nonInputProperties.systemProperty('tests.rest.cluster', "${-> testClusters."${baseName}".allHttpSocketURI.join(",")}")

